mindmap/mindmap.py

```diff
@@ -393,12 +407,38 @@ def submit_assignment(self, data, _suffix="") -> dict:
         student_item_dict = self.get_student_item_dict()
         create_submission(student_item_dict, answer)
 
-        self.submitted = True
+        self.submission_status = SubmissionStatus.SUBMITTED.value
 
         return {
             "success": True,
         }
 
+    def get_or_create_student_module(self, user):
+        """
+        Gets or creates a StudentModule for the given user for this block
+
+        Returns:
+            StudentModule: A StudentModule object
+        """
+        # pylint: disable=no-member
+        student_module, created = StudentModule.objects.get_or_create(
```

Comment:
Why are we using StudentModule objects directly?

Reply:
[edx-sga](https://github.com/mitodl/edx-sga/blob/master/edx_sga/sga.py#L723) does something similar. Could we implement it in a better way?